msteams-platform/apps-in-teams-meetings/video-filters-for-meeting-apps.md

```diff
@@ -0,0 +1,472 @@
+
+```typescript
+// notify Teams when user click on app page
+const sampleEffect1Element = document.getElementById("sampleEffect1");
+
+// add event listener for sampleEffect1 ui element. This can change according to how you implement you UI for video effect.
+
+sampleEffect1Element.addEventListener("click", function () {
+
+  // notify for selected video effect changed event
+  video.notifySelectedVideoEffectChanged("EffectChanged", effectId);
+
+});
+```
+
+### Guidelines to use the Video extensibility API
+
+Ensure that you implement the following guidelines to use the video extensibility APIs:
+
+* Video frame related data, including any raw video frame and any data calculated from video frame, must only be consumed in user's local computer, and shouldn't be uploaded to the network.
+
+* Only NV12 video format is supported. RGB(A) isn't supported. The sample app has provided sample code for video format conversion between NV12 and RGB(A).
+
+* The JavaScript in the video app must only operate on the latest videoFrame. References to previous video frames can result in unexpected behavior.
+
+* Video frame size can change anytime, so size sensitive resources for processing video frames should be recreated when video frame size changes.
+
+* Call `registerForVideoEffect()` as early as possible. Call `registerForVideoFrame()` after the required resources are downloaded and the video app initialization is finished.
+
+* Ensure the stability of the effect algorithm. When calling `notifyVideoFrameProcessed`, ensure that the video frame is fully processed even if the algorithm crashes. When using a video app, users will be surprised if they see a video frame without the effect.
+
+* Video filter app must be compliant to [Teams client SDK terms of use](/legal/microsoft-apis/terms-of-use).
+
+* Ensure that the video filter app meets the following performance requirements:
+  * Video frame processing time must be less than 30 ms.
+  * Video app loading time must be less than four seconds.
+  * App Size must be less than 20 MB.
+  * Memory size must be less than 150 MB.
+  * Latency must be less than 100 ms.
+
+## Sideload the video filter app
+
+Sideload the video filter app to your tenant in Teams. For more information, see [upload your custom app](../concepts/deploy-and-publish/apps-upload.md). After sideloading, you can use the video filter app to apply filters to videos in Teams meetings.
+
+## Diversity and Inclusion requirements
+
+The following are the guiding principles for defining requirements and app approval process:
+
+1. The brand of Microsoft as a diverse and inclusive company shouldn't be tarnished.
+1. The video filters feature should be perceived as diverse and inclusive.
+1. The filter app developer should provide filters only within their area of expertise.
+1. A filter shouldn't allow dysmorphia. For example, morphing the face, eyes, and nose, to skinnier, fatter, or distorted.
+1. A filter shouldn't allow skin smoothing.
```

Comment:
Another reviewer found this confusing because we have a defined *category* of filters "Makeup", which they observed often smooths skin by covering it up.